common/src/main/java/com/wynntils/screens/overlays/selection/widgets/OverlayButton.java

```diff
@@ -0,0 +1,305 @@
+        if (button == GLFW.GLFW_MOUSE_BUTTON_RIGHT && KeyboardUtils.isShiftDown() && overlayId != -1) {
+            deleteOverlay();
+        }
+
+        if (editInput != null && editInput.visible && editInput.mouseClicked(mouseX, mouseY, button)) {
+            return true;
+        }
+
+        if (button == GLFW.GLFW_MOUSE_BUTTON_LEFT) {
+            if (isSelected() && editInput != null) {
+                editInput.visible = true;
+                selectionScreen.setFocusedTextInput(editInput);
+            } else {
+                selectionScreen.setSelectedOverlay(overlay);
+            }
+        }
+
+        return super.mouseClicked(mouseX, mouseY, button);
+    }
+
+    @Override
+    public boolean mouseReleased(double mouseX, double mouseY, int button) {
+        // Prevent interaction when the tile is outside of the mask from the screen, same applies and released
+        if ((mouseY <= selectionScreen.getConfigMaskTopY() || mouseY >= selectionScreen.getConfigMaskBottomY())) {
+            return false;
+        }
+
+        if (editInput != null) {
+            editInput.mouseReleased(mouseX, mouseY, button);
+        }
+
+        return super.mouseReleased(mouseX, mouseY, button);
+    }
+
+    @Override
+    public void onPress() {}
+
+    @Override
+    public boolean keyPressed(int keyCode, int scanCode, int modifiers) {
+        if (keyCode == GLFW.GLFW_KEY_ENTER && editInput != null && editInput.visible) {
+            editInput.visible = false;
+
+            if (overlay instanceof CustomNameProperty customNameOverlay) {
+                customNameOverlay.setCustomName(editInput.getTextBoxInput());
+
+                if (editInput.getTextBoxInput().isEmpty()) {
+                    textToRender = overlay.getTranslatedName();
+                } else {
+                    textToRender = editInput.getTextBoxInput();
+                }
+            }
+
+            return true;
+        }
+
+        return false;
+    }
+
+    @Override
+    public void setY(int y) {
+        super.setY(y);
+
+        if (editInput != null) {
+            editInput.setY(y);
+        }
+    }
+
+    public Overlay getOverlay() {
+        return overlay;
+    }
+
+    public void hideEditInput() {
+        if (editInput != null) {
+            editInput.visible = false;
+        }
+    }
+
+    private void deleteOverlay() {
+        // Get the group holder
+        OverlayGroupHolder overlayGroupHolder = getGroupHolder();
+
+        if (overlayGroupHolder == null) {
+            WynntilsMod.error("Failed to delete, overlay group not found for overlay " + overlay.getJsonName());
+            return;
+        }
+
+        // Delete the overlay
+        Managers.Overlay.removeIdFromOverlayGroup(overlayGroupHolder, overlayId);
+
+        // Reload config
+        Managers.Config.loadConfigOptions(true, false);
+        Managers.Config.saveConfig();
```

Comment:
The other two use cases also call `ConfigManager#reloadConfig`. Why do we not that here?

Reply:
Think I just forgot it :)